File: ccif/src/java/other/org/himalay/ccif/APDUFlowConf_Detail.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class APDUFlowConf_Detail extends BinStruct implements PublicBinMsg {




    // members variables
    // flowId
    public int flowId ;
    // serviceType
    public short serviceType ;
    // ipUnicast
    public ServiceIPUnicast ipUnicast ;
    // ipMulticast
    public ServiceIPMulticast ipMulticast ;
    // socket
    public ServiceSocket socket ;

    public APDUFlowConf_Detail () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize flowId

        // Initialize serviceType

        // Initialize ipUnicast
        //Conditional, will be initialized during read
        // Initialize ipMulticast
        //Conditional, will be initialized during read
        // Initialize socket
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read flowId
        {
            flowId= BinPrimitive.readUI24(istream);
            retVal += 3;
        }
        // read serviceType
        {
            serviceType=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read ipUnicast
        if (serviceType==0x01) {
            ipUnicast = new ServiceIPUnicast();
            retVal += ipUnicast.read(istream);
        }
        // read ipMulticast
        if (serviceType==0x02) {
            ipMulticast = new ServiceIPMulticast();
            retVal += ipMulticast.read(istream);
        }
        // read socket
        if (serviceType==0x04) {
            socket = new ServiceSocket();
            retVal += socket.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        {   /** fix dependent sizes for ipUnicast **/
        }
        {   /** fix dependent sizes for ipMulticast **/
        }
        {   /** fix dependent sizes for socket **/
        }

        // write flowId
        {
            ostream.writeByte((flowId  & 0x00FF0000)>>16);
            ostream.writeShort((flowId & 0x0000FFFF));
            retVal+=3;
        }
        // write serviceType
        ostream.writeByte(serviceType);
        retVal +=1;
        // write ipUnicast
        if (ipUnicast!=null)retVal +=ipUnicast.write(ostream);
        // write ipMulticast
        if (ipMulticast!=null)retVal +=ipMulticast.write(ostream);
        // write socket
        if (socket!=null)retVal +=socket.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("APDUFlowConf_Detail\n");
        dc.increaseIndent();
        int retVal= 0;
        // write flowId
        dc.indent();
        dc.getPs().println("flowId="+flowId+"(0x"+ Integer.toHexString(flowId)+")") ;
        // write serviceType
        dc.indent();
        dc.getPs().println("serviceType="+serviceType+"(0x"+ Integer.toHexString(serviceType) + ")" );
        // write ipUnicast
        if ( ipUnicast != null ) {
            dc.indent();
            dc.getPs().println("ipUnicast") ;
            retVal +=ipUnicast.dump(dc);
        }
        // write ipMulticast
        if ( ipMulticast != null ) {
            dc.indent();
            dc.getPs().println("ipMulticast") ;
            retVal +=ipMulticast.dump(dc);
        }
        // write socket
        if ( socket != null ) {
            dc.indent();
            dc.getPs().println("socket") ;
            retVal +=socket.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for flowId
    //public int getFlowId()
    //{
    //    return flowId ;
    //}


    // Setter for flowId
    //public void setFlowId(int val)
    //{
    //    this.flowId= val;
    //}
    // Getter for serviceType
    //public short getServiceType()
    //{
    //    return serviceType ;
    //}


    // Setter for serviceType
    //public void setServiceType(short val)
    //{
    //    this.serviceType= val;
    //}
    // Getter for ipUnicast
    //public ServiceIPUnicast getIpUnicast()
    //{
    //    return ipUnicast ;
    //}


    // Setter for ipUnicast
    //public void setIpUnicast(ServiceIPUnicast val)
    //{
    //    this.ipUnicast= val;
    //}
    // Getter for ipMulticast
    //public ServiceIPMulticast getIpMulticast()
    //{
    //    return ipMulticast ;
    //}


    // Setter for ipMulticast
    //public void setIpMulticast(ServiceIPMulticast val)
    //{
    //    this.ipMulticast= val;
    //}
    // Getter for socket
    //public ServiceSocket getSocket()
    //{
    //    return socket ;
    //}


    // Setter for socket
    //public void setSocket(ServiceSocket val)
    //{
    //    this.socket= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/APDUFlowConf_Detail.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "APDUFlowConf_Detail.h"

#ifdef org_himalay_ccif_APDUFlowConf_Detail__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
APDUFlowConf_Detail::APDUFlowConf_Detail() {
    references= 0;


    // flowId
    flowId= (ui24)0;
    // serviceType
    serviceType= (ui8)0;
    // ipUnicast
    org_himalay_ccif_ServiceIPUnicast__INIT(ipUnicast)
    // ipMulticast
    org_himalay_ccif_ServiceIPMulticast__INIT(ipMulticast)
    // socket
    org_himalay_ccif_ServiceSocket__INIT(socket)
}
// Destructor
APDUFlowConf_Detail::~APDUFlowConf_Detail() {
    // flowId

    // serviceType

    // ipUnicast
    org_himalay_ccif_ServiceIPUnicast__DELETE( ipUnicast);
    // ipMulticast
    org_himalay_ccif_ServiceIPMulticast__DELETE( ipMulticast);
    // socket
    org_himalay_ccif_ServiceSocket__DELETE( socket);
}

int APDUFlowConf_Detail::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // flowId
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        flowId= i1 | i2;
        retVal += 3;
    }
    // serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // ipUnicast
    if (serviceType==0x01) {
        org_himalay_ccif_ServiceIPUnicast__NEW( ipUnicast);
        retVal += ipUnicast->read(istream);
    }
    // ipMulticast
    if (serviceType==0x02) {
        org_himalay_ccif_ServiceIPMulticast__NEW( ipMulticast);
        retVal += ipMulticast->read(istream);
    }
    // socket
    if (serviceType==0x04) {
        org_himalay_ccif_ServiceSocket__NEW( socket);
        retVal += socket->read(istream);
    }
    return retVal;
}

int APDUFlowConf_Detail::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read flowId
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        flowId= i1 | i2;
        retVal += 3;
    }
    // read serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // read ipUnicast
    if (serviceType==0x01) {
        org_himalay_ccif_ServiceIPUnicast__NEW( ipUnicast);
        retVal += ipUnicast->read(istream);
    }
    // read ipMulticast
    if (serviceType==0x02) {
        org_himalay_ccif_ServiceIPMulticast__NEW( ipMulticast);
        retVal += ipMulticast->read(istream);
    }
    // read socket
    if (serviceType==0x04) {
        org_himalay_ccif_ServiceSocket__NEW( socket);
        retVal += socket->read(istream);
    }

    return retVal;
}

int APDUFlowConf_Detail::write(DataOutputStream& ostream) { // throws IOException


    {   /** fix dependent sizes for ipUnicast **/
    }
    {   /** fix dependent sizes for ipMulticast **/
    }
    {   /** fix dependent sizes for socket **/
    }

    int retVal= 0;
    // write flowId
    {
        ostream.writeUI8((flowId  & 0x00FF0000)>>16);
        ostream.writeUI16((flowId & 0x0000FFFF));
        retVal+=3;
    }
    // write serviceType
    ostream.writeUI8(serviceType);
    retVal +=1;
    // write ipUnicast
    if ( ipUnicast != NULL ) {
        retVal +=ipUnicast->write(ostream);
    }
    // write ipMulticast
    if ( ipMulticast != NULL ) {
        retVal +=ipMulticast->write(ostream);
    }
    // write socket
    if ( socket != NULL ) {
        retVal +=socket->write(ostream);
    }

    return retVal;
}

int APDUFlowConf_Detail::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("APDUFlowConf_Detail\n");
    dc.increaseIndent();
    int retVal= 0;
    // write flowId
    dc.indent();
    dc.getPs().print("flowId=");
    dc.getPs().println((long)flowId);
    // write serviceType
    dc.indent();
    dc.getPs().print("serviceType=");
    dc.getPs().println((long)serviceType);
    // write ipUnicast
    if ( ipUnicast != NULL ) {
        dc.indent();
        dc.getPs().println("ipUnicast") ;
        retVal +=ipUnicast->dump(dc);
    }
    // write ipMulticast
    if ( ipMulticast != NULL ) {
        dc.indent();
        dc.getPs().println("ipMulticast") ;
        retVal +=ipMulticast->dump(dc);
    }
    // write socket
    if ( socket != NULL ) {
        dc.indent();
        dc.getPs().println("socket") ;
        retVal +=socket->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for flowId
//ui24 APDUFlowConf_Detail::getFlowId()
//{
//return this->flowId;
//}

// Setter for flowId
//void APDUFlowConf_Detail::setFlowId(ui24 val)
//{
//this->flowId= val;
//}

// Getter for serviceType
//ui8 APDUFlowConf_Detail::getServiceType()
//{
//return this->serviceType;
//}

// Setter for serviceType
//void APDUFlowConf_Detail::setServiceType(ui8 val)
//{
//this->serviceType= val;
//}

// Getter for ipUnicast
//org_himalay_ccif_ServiceIPUnicast__PTR_TYPE APDUFlowConf_Detail::getIpUnicast()
//{
//return this->ipUnicast;
//}

// Setter for ipUnicast
//void APDUFlowConf_Detail::setIpUnicast(org_himalay_ccif_ServiceIPUnicast__PTR_TYPE val)
//{
//this->ipUnicast= val;
//}

// Getter for ipMulticast
//org_himalay_ccif_ServiceIPMulticast__PTR_TYPE APDUFlowConf_Detail::getIpMulticast()
//{
//return this->ipMulticast;
//}

// Setter for ipMulticast
//void APDUFlowConf_Detail::setIpMulticast(org_himalay_ccif_ServiceIPMulticast__PTR_TYPE val)
//{
//this->ipMulticast= val;
//}

// Getter for socket
//org_himalay_ccif_ServiceSocket__PTR_TYPE APDUFlowConf_Detail::getSocket()
//{
//return this->socket;
//}

// Setter for socket
//void APDUFlowConf_Detail::setSocket(org_himalay_ccif_ServiceSocket__PTR_TYPE val)
//{
//this->socket= val;
//}

int APDUFlowConf_Detail::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_APDUFlowConf_Detail__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::APDUFlowConf_Detail* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::APDUFlowConf_Detail* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code